### ISSUE TITLE
Feature Request: Flatten directories like github.

### ISSUE THREAD

**Issue by GordianDziwis** [ranger/ranger#1257]
I am working with java and the sources directories are very deep. Dynamic flattening would be a great help,  [here](https://github.com/eclipse/eclipse.jdt.ls/tree/master/org.eclipse.jdt.ls.tests) is an example.

**Comment by toonn** [comment]
Hmm, so flatten up to the point where there's no longer just a single directory in a directory?

**Comment by toonn** [comment]
Don't see how it would? I'm currently thinking about having move right automatically move right again in the single subdir case. So only "previews" of directories would have to be changed too.

**Comment by BonaBeavis** [comment]
Yes you are right, it would only the performance problems with `flat` if this replaces the old `flat`. And yes only the "preview" needs to be changed.

Maybe a limit would help the flat command, start with `flat = 1` and count the number of directories/files and check if the number is smaller than the limit. If it does: increment to `flat = 1 + 1` and check again, if it exceeds just stop.

**Comment by toonn** [comment]
Not sure what problems you're experiencing with flatten, I regularly flatten a directory with more than 1000 files. I've noticed pypy really speeds up ranger in large directories, maybe try that if you want more speed.

This would in no way replace the flatten command, the behavior is too different to even reuse much of flatten's implementation.
